pkg/kube/client.go

```diff
@@ -104,6 +104,23 @@ func (c *Client) Create(namespace string, reader io.Reader, timeout int64, shoul
 		if info.Namespace != namespace && info.Namespace != "" {
 			return fmt.Errorf("resource's namespace %s doesn't match the current namespace %s", info.Namespace, namespace)
 		}
+		if name != "" {
+			raw := info.Object.(runtime.Unstructured).UnstructuredContent()
+			label, ok := GetValue(raw, "metadata", "labels")
+			if ok {
+				if v, ok := label.(map[string]interface{}); ok {
+					v["io.cattle.field/appId"] = name
```

Comment:
Two things about this annotation:
1. The format for "field" annotations is `field.cattle.io/projectId`, not `io.cattle.field`
2. Using `field.cattle.io` gives it special meaning it means that norma (and the rancher api) should interpret it as an API field (or at least thats one step in exposing an annotation as a field). Was that your intention?